packages/api/amplify_api/lib/src/amplify_authorization_rest_client.dart

```diff
@@ -50,8 +52,12 @@ class AmplifyAuthorizationRestClient extends http.BaseClient
   http.BaseRequest _authorizeRequest(http.BaseRequest request) {
     if (!request.headers.containsKey(AWSHeaders.authorization) &&
         endpointConfig.authorizationType != APIAuthorizationType.none) {
-      // ignore: todo
-      // TODO: Use auth providers from core to transform the request.
+      // TODO(ragingsquirrel3): Use auth providers from core to transform the request.
+      final apiKey = endpointConfig.apiKey;
+      if (endpointConfig.authorizationType == APIAuthorizationType.apiKey &&
+          apiKey != null) {
+        request.headers.putIfAbsent(_xApiKey, () => apiKey);
+      }
```

Comment:
I would rearrange this a bit. If the authorization type is API key but there is no API key we should alert the user (throw an error)